### PR TITLE
Preserve undo history and select child meshes recursively

### LINKED ISSUE #17: Export FBX clears my Blender undo history
When executing the Export FBX function within Chiro UE4, I lose all of my Undo History. The title bar shows that I have a `(Recovered)` blend file and my Undo History just has two entries: `Original` and `Export FBX (Chiro UE4)`. I expect the plugin to not change my prior Undo History and for the session to become "dirty" because I made the export function, rather than appearing to load a recovered save?

I tested the normal FBX export function in Blender, and it does not clear my Undo History

Blender version: v2.92.0
Chiro UE4 version: v0.1.1

### PR DESCRIPTION
**Pull request by mikeseese**
This PR accomplishes two things:
1. Fixes #17 by preserving undo history when exporting via FBX by creating a ephemeral duplicate armature, making the modifications on the ephemeral armature, exporting that, and finally deleting it. During this I changed some logic to rename the existing `root` armature to `root.XXX` to follow suit with the rest of blender renaming, even though it doesn't really matter. I did this under the impression a second export would conflict, but it shouldn't. I didn't remove it because it's safer in case a user has a `root-renamed` object.
1. My armature has meshes as children of `EMPTY` objects for organizational purposes, so this also recursively selects child meshes.

This PR was tested on my use case, and I unfortunately cannot spend more time testing this. Please do the due diligence to test further. If you're not interested in the changes without more testing, no biggie; I'll just close the PR. Sorry I can't put more TLC into this.

### CODE REVIEW

**Comment by dnsl48** [comment]
Thank you for the patch!
It is definitely worth investigating as an alternative to our current "recovery" approach!
I'll find some time to test it with our other use cases :+1:

**Comment by dnsl48** [comment]
Thank you for the update @seesemichaelj 
Do you reckon that's the final version, or you're considering any more patches to this PR? :)

**Comment by mikeseese** [comment]
I believe this is my final version! Thanks for checking

File: addon/ops/fbx_export.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import bpy
+import random
```
Comment (dnsl48) [inline addon/ops/fbx_export.py:2]: looks like an unused import here?

**Review by dnsl48 — APPROVED** [review]
That looks good!

Sorry for the delay. Apart from a redundant import this seems solid!
Thank you for the contribution!

**Comment by dnsl48** [comment]
Released in version `0.1.2`
Thanks again for the contribution! :)

**Comment by mikeseese** [comment]
Glad to help!

**Pull request merged**